action.yml

```diff
@@ -1,139 +1,161 @@
+        else
+          echo "Checking releases"
+          GIT_RELS_JSON=$(wget --header="Authorization: Bearer ${{ github.token }}" -qO- "https://api.github.com/repos/${{ inputs.repository }}/releases")
+          GIT_RELS=$(echo "$GIT_RELS_JSON" | jq -r '.[].tag_name' | sort -V)
+          GIT_NPRS=$(echo "$GIT_RELS_JSON" | jq -r '.[] | select((.draft == false) and (.prerelease == false)) | .tag_name' | sort -V)
+
+          # Add "v" in front if missing
+          REL=${{ inputs.gap-version }}
+          REL=v${REL#v}
+          
+          if echo "$GIT_RELS" | grep -qx "$REL" ; then
+            echo "Version: exact release match"
+            IS_RELEASE=true
+            VERSION=$REL
+          elif echo "$GIT_NPRS" | grep -q "^$REL\." ; then
+            echo "Version: expanded to release"
+            IS_RELEASE=true
+            VERSION=$(echo "$GIT_NPRS" | grep "^$REL\." | tail -n 1)
+          else
+            echo "Checking branches and tags"
+            if git ls-remote --heads https://github.com/${{ inputs.repository }}.git | sed 's|.*refs/heads/||' | grep -qx "${{ inputs.gap-version }}" ; then
+              echo "Version: exact branch match"
+              VERSION=${{ inputs.gap-version }}
+            elif git ls-remote --tags --refs https://github.com/${{ inputs.repository }}.git | sed 's|.*refs/tags/||' | grep -qx "${{ inputs.gap-version }}" ; then
+              echo "Version: exact tag match"
+              VERSION=${{ inputs.gap-version }}
             else
-              packages=(
-                libgmp-dev
-                libreadline-dev
-                zlib1g-dev
-              )
-            fi
-            sudo apt-get install "${packages[@]}"
-          elif [ "${{runner.os}}" = "macOS" ]; then
-            if [ "${{inputs.ABI}}" = "32" ]; then
-              echo "Can't use macOS with 32bit!"
+              echo "No release, branch or tag with name ${{ inputs.gap-version }} found"
               exit 1
             fi
-            brew install gmp zlib autoconf && brew link zlib
           fi
-        shell: bash
+        fi
 
-      - name: "Clone GAP"
-        shell: bash
-        run: git clone --depth=2 -b ${{ inputs.GAPBRANCH }} https://github.com/gap-system/gap.git $HOME/gap
+        echo "VERSION=$VERSION" >> "$GITHUB_OUTPUT"
+        echo "IS_RELEASE=$IS_RELEASE" >> "$GITHUB_OUTPUT"
 
-      - name: "Build GAP"
-        shell: bash
-        env:
-          ABI: ${{ inputs.ABI }}
-        run: |
-          cd $HOME/gap
-          CONFIGFLAGS="${{ inputs.CONFIGFLAGS }}"
-          # for HPC-GAP, add suitable flags
-          if [[ ${{ inputs.HPCGAP }} = yes ]]; then
-            CONFIGFLAGS="$CONFIGFLAGS --enable-hpcgap"
-          fi
+    - name: "Set GAPROOT"
+      shell: bash
+      run: |
+        GAPROOT="$HOME/gap"
+        mkdir -p $GAPROOT
+        echo "GAPROOT=$GAPROOT" >> "$GITHUB_ENV"
```

Comment:
Currently also `build-pkg-docs` sets `GAPROOT` in `$GITHUB_ENV` like that. But of course it "belongs" into `setup-gap`. So as a note to self: the next breaking version of `build-pkg-docs` probably should do away with setting `GAPROOT` (and `GAP`) in the environment.